sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/PartitionKeyRangeServerBatchRequest.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.batch;
+
+import com.azure.cosmos.CosmosItemOperation;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public final class PartitionKeyRangeServerBatchRequest extends ServerBatchRequest {
+
+    private final String partitionKeyRangeId;
```

Comment:
PartitionKeyRangeServerBatchRequest at some point will be translated to a RxDocumentServiceRequest. right? how does partition split proofing work?

PartitionKeyRangeServerBatchRequest may contain multiple documents. we create one RxDocumentServiceRequest out of all those documents. now if a partition split happens we cannot just retry the same RxDocumentServiceRequest as some documents may go to a left child some to the right child partition. how are we handling that scenario with retries?

Reply:
+1 - we need a separate retry policy that will cover splitting the individual operations into the "right" partition. I assume this means we need to back-channel documents impacted into a GoneRetryQueue (which is handled in parallel with the incoming flux of operations). Bulk executor library is handling this similarly...

Reply:
Have handled it using a sink, which will re-evaluate the partition key range id.

Reply:
@rakkuma we need tests for partition split. please add TODO and create github item to track this.
without partition split support we cannot announce bulk.

1. mocked unit tests: see `DocumentProducerTest` as a mocked unit tests for partition split on quer.
2. end to end integration tests. see `ReadMyWritesConsistencyTest` as a e2e tests which tests partition split. essentially some of our CIs target prod, you can scale the throughput of the cosmos container which will result in partition split. 

I think tests for partition split and its implementation require more work. please capture this work after this PR gets in.

Reply:
Partition split is handled already. Yeah the test case is remaining. I will try to add those in this PR only. Thanks.